modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenOperation.java

```diff
@@ -12,7 +12,7 @@
 
 public class CodegenOperation {
     public final List<CodegenProperty> responseHeaders = new ArrayList<CodegenProperty>();
-    public boolean hasAuthMethods, hasConsumes, hasProduces, hasParams, hasOptionalParams,
+    public boolean hasAuthMethods, hasConsumes, hasProduces, hasParams, hasOptionalParams, hasRequiredParams,
```

Comment:
I'm surprised this is needed. Most generator use the params array and loop with "isRequired" to build almost everything. Is there any specific reason to add this to the "generic code"?

Reply:
Yeah, you can check `{{isRequired}}` but that's not sufficient, because we use `{{hasNext}}` to see if there's a subsequent element, and that isn't aware of `hasRequired`, so it was generating code with 2 commas in a row.

Reply:
Ok makes sense!